tests/pausing_test.go

```diff
@@ -363,30 +364,32 @@ var _ = Describe("[rfe_id:3064][crit:medium][vendor:cnv-qe@redhat.com][level:com
 
 	Context("A long running process", func() {
 
+		grepSleepPid := func(expecter expect.Expecter) string {
+			res, err := tests.ExpectBatchWithValidatedSend(expecter, []expect.Batcher{
+				&expect.BSnd{S: `pgrep -f "sleep 5"` + "\n"},
+				&expect.BExp{R: tests.Retcode("[0-9]+", "\\# ")}, // pid
+			}, 15*time.Second)
+			log.DefaultLogger().Infof("a:%+v\n", res)
+			Expect(err).ToNot(HaveOccurred())
+			re := regexp.MustCompile("\r\n[0-9]+\r\n")
+			return strings.TrimSpace(re.FindString(res[0].Match[0]))
+		}
+
 		startProcess := func(expecter expect.Expecter) string {
 			By("Start a long running process")
-			res, err := expecter.ExpectBatch([]expect.Batcher{
+			res, err := tests.ExpectBatchWithValidatedSend(expecter, []expect.Batcher{
 				&expect.BSnd{S: "sleep 5&\n"},
 				&expect.BExp{R: "\\# "}, // prompt
-				&expect.BSnd{S: `pgrep -f "sleep 5"` + "\n"},
-				&expect.BExp{R: "sleep 5"},    // command
-				&expect.BExp{R: "[0-9]+\r\n"}, // pid
 			}, 15*time.Second)
 			log.DefaultLogger().Infof("a:%+v\n", res)
 			Expect(err).ToNot(HaveOccurred())
-			return strings.TrimSpace(res[2].Match[0])
+
+			return grepSleepPid(expecter)
 		}
 
 		checkProcess := func(expecter expect.Expecter) string {
 			By("Checking the long running process")
-			res, err := expecter.ExpectBatch([]expect.Batcher{
-				&expect.BSnd{S: `pgrep -f "sleep 5"` + "\n"},
-				&expect.BExp{R: "sleep 5"},    // command
-				&expect.BExp{R: "[0-9]+\r\n"}, // pid
-			}, 15*time.Second)
-			log.DefaultLogger().Infof("a:%+v\n", res)
-			Expect(err).ToNot(HaveOccurred())
-			return strings.TrimSpace(res[1].Match[0])
+			return grepSleepPid(expecter)
```

Comment:
I think you should return the output of the command and do the assert and pid extraction here. Otherwise the name of the func does  not represent well what is going on (and it will be nice to separate the operation from the assertion).

Reply:
need to consume the prompt at grepSleepPid as well ?

Reply:
@oshoval this is an internal method for a very specific purpose. The prompt is known, there is no need to pass it as a parameter.

Reply:
i meant to consume it, not the pass it

Reply:
@EdDev the pupose of this internal method was reuse. This method as its name specifies does grep to get the sleep pid. The method is internal and not a tool that can be resued in other places that may want to have different assersion. I think it is fine and convenient to have the assertion inside it.

Reply:
> i meant to consume it, not the pass it

Sorry, I don't understand the comment.
`grepSleepPid` waits for the prompt. What else do you think it should do?

Reply:
if it Exp for the prompt at the end of it, then all good

Reply:
> I think it is fine and convenient to have the assertion inside it.

I do not like exactly this inner assertion.
It gives bad tracing and usually can be "fxied" by using an offset-assertion which is not well maintained (as it depends the stack through which it is called).
Inner assertions also gives less control.

Anyway, I will not insist on this.